bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/HttpServiceUtil.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.net;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * Some utility functions related to the http service
+ *
+ * @author Markus Rathgeb - Initial contribution and API
+ */
+public class HttpServiceUtil {
+
+    private HttpServiceUtil() {
+    }
+
+    /**
+     * Get the port that is used by the HTTP service.
+     *
+     * @param bc the bundle context used for lookup
+     * @return the port if used, -1 if no port could be found
+     */
+    public static int getHttpServicePort(final BundleContext bc) {
+        return getHttpServicePortProperty(bc, "org.osgi.service.http.port");
+    }
+
+    public static int getHttpServicePortSecure(final BundleContext bc) {
+        return getHttpServicePortProperty(bc, "org.osgi.service.http.port.secure");
+    }
+
+    // Utility class that could be used for non-secure and secure port.
```

Comment:
The logic for "org.osgi.service.http.port" and "org.osgi.service.http.port.secure" is the same only the property name differs. So a private method `getHttpServicePortProperty` has been used that takes the property as argument.